packages/browserless/src/index.js

```diff
@@ -38,7 +39,13 @@ module.exports = ({
       ...launchOpts
     })
 
-    browser.on('disconnected', createBrowser)
+    browser.on('disconnected', () => {
+      setTimeout(async () => {
+        const pid = browser.process().pid
+        await fkill(pid)
+        createBrowser()
```

Comment:
Maybe we can move this out of the `setTimeout` so we don't wait to create a new browser?

Reply:
I'm not totally sure – the point with `setTimeout` is to avoid a race condition, performing the action in the next clock cycle